dags/prio/processor.py

```diff
@@ -0,0 +1,330 @@
+"""Airflow configuration for coordinating the Prio processing pipeline.
+
+The prio-processor docs details the internals of the containerized application.
+It contains storage and processing scripts that are coordinated by Airflow.
+
+The graph has three phases: partitioning, publishing, and processing.
+Partitioning reads batched prio pings and range-partitions across date,
+batch-id, and share-id. The partitioned data sets are published to all
+subscribed servers. Each of the servers runs a Kubernetes cluster running from
+the `mozilla/prio-processor:latest` image. A container runs the processing
+pipeline, starting with input at `{private-bucket}/raw/` and resulting in output
+at `{private-bucket}/processed/`.
+
+There are two sets of servers that run in Google Kubernetes Engine (GKE) under
+Data Operations. The servers are assigned service accounts with access to a
+private storage bucket associated with its project, and a pair of mutually
+shared buckets. An administrative account oversees interop with ingestion,
+including a process and transfer job from Firefox Telemetry into a receiving
+bucket located in each server's project.
+
+The SubDag operator simplifies the definition of each phase. The clusters are
+ephemeral -- there are situations where clusters can become orphaned. These are
+usually resolved by rerunning the individual SubDag, which should clean-up
+properly.
+
+The following Airflow variables should be set:
+
+    prio_private_key_hex_internal
+    prio_public_key_hex_internal
+    prio_private_key_hex_external
+    prio_public_key_hex_external
+    prio_shared_secret
+
+These variables are encrypted and passed into the prio-processor container via
+the environment.
+"""
+from datetime import datetime, timedelta
+from os import environ
+
+from airflow import DAG
+from airflow.contrib.hooks.gcs_hook import GoogleCloudStorageHook
+from airflow.contrib.operators.gcs_to_gcs import (
+    GoogleCloudStorageToGoogleCloudStorageOperator,
```

Comment:
list not required

Reply:
This is enforced by black due to the length of the line.